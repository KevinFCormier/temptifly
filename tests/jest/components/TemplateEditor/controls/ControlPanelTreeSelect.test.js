'use strict'

import React from 'react'
import ControlPanelTreeSelect from '../../../../../src/controls/ControlPanelTreeSelect'
import renderer from 'react-test-renderer'

import { control } from '../../TestingData'

describe('ControlPanelTreeSelect component', () => {
  it('renders as expected', () => {
    const fn = jest.fn()
    const component = renderer.create(
      <ControlPanelTreeSelect
        key={'key'}
        control={control}
        controlId={'controlId'}
        handleChange={fn}
        i18n={fn}
      />
    )
    expect(component.toJSON()).toMatchSnapshot()
  })
})
