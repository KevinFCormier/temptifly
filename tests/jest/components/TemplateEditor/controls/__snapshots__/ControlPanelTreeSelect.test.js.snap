// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ControlPanelTreeSelect component renders as expected 1`] = `
<div
  className="creation-view-controls-treeselect"
>
  <div
    className="pf-c-form__group"
    id="controlId-label"
  >
    <div
      className="pf-c-form__group-label"
    >
      <label
        className="pf-c-form__label"
        htmlFor="controlId"
      >
        <span
          className="pf-c-form__label-text"
        >
          creation.app.name
        </span>
      </label>
       
      <button
        aria-label="More info"
        className="pf-c-form__group-label-help"
        id="controlId-label-help-button"
        onClick={[Function]}
      >
        <svg
          aria-hidden={true}
          aria-labelledby={null}
          fill="currentColor"
          height="1em"
          role="img"
          style={null}
          viewBox="0 64 1024 1024"
          width="1em"
        >
          <path
            d="M512.059-73.143c-282.338 0-512.059 229.673-512.059 512.025 0 282.235 229.721 511.975 512.059 511.975 282.281 0 511.941-229.735 511.941-511.975 0.005-282.352-229.659-512.025-511.941-512.025zM512.059 826.523c-213.826 0-387.728-173.856-387.728-387.643 0-213.89 173.904-387.694 387.728-387.694 213.717 0 387.671 173.803 387.671 387.694 0.005 213.785-173.957 387.643-387.671 387.643zM585.143 164.571v109.714c0 4.951-1.808 9.237-5.429 12.857s-7.906 5.429-12.857 5.429h-109.714c-4.953 0-9.239-1.808-12.857-5.429s-5.429-7.906-5.429-12.857v-109.714c0-4.951 1.81-9.237 5.429-12.857s7.904-5.429 12.857-5.429h109.714c4.951 0 9.237 1.808 12.857 5.429s5.429 7.906 5.429 12.857zM521.616 365.714c118.343 0 214.286 88.965 214.286 187.429s-95.943 178.286-214.286 178.286c-173.045 0-208.091-93.714-213.963-171.113 0-7.936 6.729-9.838 15.872-9.838s108.073 0 113.143 0c6.096 0 14.475 0.633 17.554 10.571 0 48.857 135.968 54.953 135.968-7.904 0-31.506-29.717-63.817-68.571-66.194s-82.286-7.607-82.286-54.199c0-13.022 0-25.673 0-44.693 0-19.015 9.717-22.343 27.431-22.343s54.853-0.002 54.853-0.002z"
            transform="rotate(180 0 512) scale(-1 1)"
          />
        </svg>
      </button>
    </div>
    <div
      className="pf-c-form__group-control"
    >
      <div
        id="controlId"
      >
        <div
          aria-label="Choose an item"
          className="tf--list-box"
          role="listbox"
          tabIndex="0"
        >
          <div
            aria-expanded={false}
            aria-haspopup="true"
            aria-label="Open menu"
            className="pf-c-form-control input"
            data-toggle="true"
            onClick={[Function]}
            onKeyPress={[Function]}
            role="button"
            tabIndex="0"
            type="button"
          >
            <input
              aria-controls="controlId-creation.app.name-"
              aria-expanded="true"
              aria-label="ListBox input field"
              autoComplete="new-password"
              className="pf-c-form-control"
              id="controlId-input"
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              placeholder=""
              role="combobox"
              spellCheck="false"
              value="test"
            />
            <div
              className="tf--list-box__selection"
              onClick={[Function]}
              onKeyPress={[Function]}
              role="button"
              tabIndex="0"
              title="Clear selected item"
            >
              <svg
                aria-hidden={true}
                aria-labelledby={null}
                fill="currentColor"
                height="1em"
                role="img"
                style={
                  Object {
                    "verticalAlign": "-0.125em",
                  }
                }
                viewBox="0 0 512 512"
                width="1em"
              >
                <path
                  d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm121.6 313.1c4.7 4.7 4.7 12.3 0 17L338 377.6c-4.7 4.7-12.3 4.7-17 0L256 312l-65.1 65.6c-4.7 4.7-12.3 4.7-17 0L134.4 338c-4.7-4.7-4.7-12.3 0-17l65.6-65-65.6-65.1c-4.7-4.7-4.7-12.3 0-17l39.6-39.6c4.7-4.7 12.3-4.7 17 0l65 65.7 65.1-65.6c4.7-4.7 12.3-4.7 17 0l39.6 39.6c4.7 4.7 4.7 12.3 0 17L312 256l65.6 65.1z"
                  transform=""
                />
              </svg>
            </div>
            <div
              className="tf--list-box__menu-icon"
              onClick={[Function]}
              onKeyPress={[Function]}
              role="button"
              tabIndex="0"
            >
              <svg
                alt="Open menu"
                aria-label="Open menu"
                fillRule="evenodd"
                height="5"
                role="img"
                viewBox="0 0 10 5"
                width="10"
              >
                <title>
                  Close menu
                </title>
                <path
                  d="M0 0l5 4.998L10 0z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
      
    </div>
  </div>
</div>
`;
